### PR TITLE
Fix filesystem linkage with libc++ version 9 or later

### LINKED ISSUE #180: fails to build on Focal with clang/libcxx
See https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/358/consoleFull#console-section-208

### PR DESCRIPTION
**Pull request by rotu**
fix #180

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
Pushed to the ros fork as the `fix-180` branch for CI.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9737/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5370/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7918/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9611/)
* Linux clang [Build Status](https://ci.ros2.org/job/ci_linux/9736/)

File: pluginlib/CMakeLists.txt [hunk 1]
```diff
@@ -33,7 +33,9 @@ install(
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
 
-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_FLAGS MATCHES "-stdlib=libc\\+\\+")
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND 9 VERSION_LESS_EQUAL CMAKE_CXX_COMPILER_VERSION)
```
Comment (dirk-thomas) [inline pluginlib/CMakeLists.txt]: `VERSION_LESS_EQUAL` doesn't exist in CMake 3.5 which is the minimum version of this CMake file.

Comment (rotu) [reply pluginlib/CMakeLists.txt]: Thanks! good catch! This seems like an ideal thing for CI to be checking...

**Comment by nuclearsandwich** [comment]
@rotu a very similar change was also proposed as #183 which should replace this PR. I'm going to close this one for now but if there's something that I'm missing feel free to re-open and rebase. In either case I appreciate the PR.

**Pull request closed without merging**